File: Clientes.py
class Clientes:
    def __init__(self, nombre, apellido, id, fecha_de_registro):
        self.nombre = nombre
        self.apellido = apellido
        self.fecha_de_registro = fecha_de_registro
        self.productos_comprados = []

    def registrar_cliente(self, nombre, apellido, id_cliente, fecha_de_registro):
        nuevo_cliente = Clientes(nombre, apellido, id_cliente, fecha_de_registro)
        lista_de_clientes.append(nuevo_cliente)

    def comprar_producto(self, producto, cantidad):
        if producto.stock >= cantidad:
            producto.stock -= cantidad
            self.productos_comprados.append(producto)
            print("Compra realizada correctamente.")
        else:
            print("No hay suficiente stock del producto.")

    def mostrar_lista_clientes(self):
        print("Lista de clientes:")
        for cliente in lista_de_clientes:
            print("Nombre:", cliente.nombre, ", Apellido:", cliente.apellido, ", ID:", cliente.id)

    def mostrar_info_cliente(self, id_cliente):
        for cliente in lista_de_clientes:
            if cliente.id == id_cliente:
                print("Nombre:", cliente.nombre)
                print("Apellido:", cliente.apellido)
                print("Fecha de Registro:", cliente.fecha_de_registro)
                print("Productos Comprados:")
                for producto in cliente.productos_comprados:
                    print("   -", producto.nombre)
                return
        print("Cliente no encontrado.")

    def eliminar_cliente(self, id_cliente):
        for cliente in lista_de_clientes:
            if cliente.id == id_cliente and not cliente.productos_comprados:
                lista_de_clientes.remove(cliente)
                print("Cliente eliminado correctamente.")
                return
        print("No se puede eliminar el cliente.")

    def obtener_numero_total_clientes(self):
        return len(lista_de_clientes)


# Lista de clientes
lista_de_clientes = []

File: src/Clientes.java
import java.util.ArrayList;
import Productos.Productos;

public class Clientes {
    public ArrayList<Clientes> getListaDeClientes() {
        return listaDeClientes;
    }

    ArrayList <Clientes> listaDeClientes=new ArrayList<>();
    private String nombre;
    private String apellido;
    private String fechaDeRegistro;
    private ArrayList<Productos> productosComprados;

    public int getId() {
        return id;
    }

    private int id;

    public Clientes(String nombre, String apellido, int id,String fechaDeRegistro){
        this.nombre=nombre;
        this.apellido=apellido;
        this.fechaDeRegistro=fechaDeRegistro;
        productosComprados = new ArrayList<>();

    }
    public void registrarclientes(String nombre,String apellido,int id, String fechaDeRegistro){
        Clientes nuevoCliente=new Clientes(nombre, apellido, id, fechaDeRegistro);
        listaDeClientes.add(nuevoCliente);
    }
    public void comprarProducto(Productos producto, int cantidad) {
        // Verificar si hay suficiente stock
        if (producto.getStock() >= cantidad) {
            // Reducir el stock del producto
            producto.setStock(producto.getStock() - cantidad);
            // Agregar el producto a la lista de productos comprados por el cliente
            productosComprados.add(producto);
            System.out.println("Compra realizada correctamente.");
        } else {
            System.out.println("No hay suficiente stock del producto.");
        }
    }

    // Método para mostrar la información básica de los clientes (nombre, apellido e id)
    public void mostrarListaClientes() {
        System.out.println("Lista de clientes:");
        for (Clientes cliente : listaDeClientes) {
            System.out.println("Nombre: " + cliente.nombre + ", Apellido: " + cliente.apellido + ", ID: " + cliente.id);
        }
    }

    // Método para mostrar la información completa de un cliente específico por su id
    public void mostrarInfoCliente(int idCliente) {
        for (Clientes cliente : listaDeClientes) {
            if (cliente.id == idCliente) {
                System.out.println("Nombre: " + cliente.nombre);
                System.out.println("Apellido: " + cliente.apellido);
                System.out.println("Fecha de Registro: " + cliente.fechaDeRegistro);
                System.out.println("Productos Comprados:");
                for (Productos producto : cliente.productosComprados) {
                    System.out.println("   - " + producto.getNombre());
                }
                return;
            }
        }
        System.out.println("Cliente no encontrado.");
    }

    // Método para eliminar un cliente por su id si no ha comprado ningún producto
    public void eliminarCliente(int idCliente) {
        for (int i = 0; i < listaDeClientes.size(); i++) {
            Clientes cliente = listaDeClientes.get(i);
            if (cliente.id == idCliente && cliente.productosComprados.isEmpty()) {
                listaDeClientes.remove(i);
                System.out.println("Cliente eliminado correctamente.");
                return;
            }
        }
        System.out.println("No se puede eliminar el cliente.");
    }

    // Método para obtener el número total de clientes
    public int obtenerNumeroTotalClientes() {
        return listaDeClientes.size();
    }
}